configuration-service/handlers/service_resource.go

```diff
@@ -208,15 +208,20 @@ func PostProjectProjectNameStageStageNameServiceServiceNameResourceHandlerFunc(
 
 	for _, res := range params.Resources.Resources {
 		filePath := serviceConfigPath + "/" + *res.ResourceURI
-		logger.Debug("Adding resource: " + filePath)
-
-		common.WriteBase64EncodedFile(filePath, res.ResourceContent)
 
 		if strings.Contains(filePath, "helm") && strings.HasSuffix(*res.ResourceURI, ".tgz") {
 			if resp := untarHelm(res, filePath); resp != nil {
 				return resp
 			}
 		}
+
+		logger.Debug("Adding resource: ", filePath)
+		err := common.WriteBase64EncodedFile(filePath, res.ResourceContent)
+		if err != nil {
+			logger.WithError(err).Errorf("Could not add resource %s to project %s", filePath, params.ProjectName)
+			return service_resource.NewPostProjectProjectNameStageStageNameServiceServiceNameResourceBadRequest().
+			WithPayload(&models.Error{Code: 400, Message: swag.String("Could not add resource")})
+		}
```

Comment:
I would wrap this code into a function since it is duplicated at line 345 and you can accept the action (update/add) as a parameter

Reply:
will make, thanks